### ISSUE TITLE
Silent semantic mismatch due to unsupported axis configuration in TFLite L2 normalization

### ISSUE THREAD

**Issue by beanduan22** [tensorflow/tensorflow#116049]
```
import io
import sys

try:
    import numpy as np
    import tensorflow as tf
    import torch
    import onnxruntime as ort
except ImportError as e:
    print(f"missing dep: {e}", file=sys.stderr)
    sys.exit(2)


def numpy_ref(x):
    """Whole-tensor L2 normalize, matching tf.math.l2_normalize(axis=None)."""
    return x / np.linalg.norm(x)


def run_keras(x_np):
    class M(tf.keras.Model):
        def __init__(self):
            super(M, self).__init__()
        def call(self, x):
            return tf.math.l2_normalize(x)
    m = M()
    return m(tf.constant(x_np)).numpy(), m


def run_xla(x_np):
    @tf.function(jit_compile=True, input_signature=[tf.TensorSpec(x_np.shape, tf.float32)])
    def f(x):
        return tf.math.l2_normalize(x)
    return f(tf.constant(x_np)).numpy()


def run_torch(x_np):
    xt = torch.tensor(x_np)
    return (xt / torch.sqrt(torch.sum(xt * xt))).numpy()


def run_onnxrt(x_np):
    class TM(torch.nn.Module):
        def forward(self, x):
            return x / torch.sqrt(torch.sum(x * x))
    buf = io.BytesIO()
    torch.onnx.export(TM(), (torch.tensor(x_np),), buf,
                      input_names=["x"], output_names=["y"], opset_version=17)
    sess = ort.InferenceSession(buf.getvalue(), providers=["CPUExecutionProvider"])
    return sess.run(None, {"x": x_np})[0]


def run_tflite(keras_model, x_np):
    tfl = tf.lite.TFLiteConverter.from_keras_model(keras_model).convert()
    itp = tf.lite.Interpreter(model_content=tfl)
    in_idx = itp.get_input_details()[0]["index"]
    itp.resize_tensor_input(in_idx, list(x_np.shape))
    itp.allocate_tensors()
    itp.set_tensor(in_idx, x_np)
    itp.invoke()
    return itp.get_tensor(itp.get_output_details()[0]["index"])


CASES = [
    ("Case 3: shape=(2,2,3), half-step floats",
     (np.arange(12, dtype=np.float32) * 0.5 + 1.0).reshape(2, 2, 3)),
    ("Case 4: shape=(4,1), unit innermost dim",
     np.array([[2.], [3.], [4.], [5.]], dtype=np.float32)),
]

any_bug = False
for label, X in CASES:
    print("=" * 78)
    print(f"{label}    shape={tuple(X.shape)}")
    print(f"  input    : {X.flatten().tolist()}")
    expected = numpy_ref(X)
    print(f"  expected : {[round(float(v), 4) for v in expected.flatten()]}")

    keras_out, m = run_keras(X)
    xla_out      = run_xla(X)
    torch_out    = run_torch(X)
    ort_out      = run_onnxrt(X)
    tflite_out   = run_tflite(m, X)

    rows = [
        ("Keras eager",  keras_out),
        ("XLA",          xla_out),
        ("PyTorch",      torch_out),
        ("ONNX Runtime", ort_out),
        ("TFLite",       tflite_out),
    ]
    print(f"  {'Runtime':<14} {'Output (rounded 4dp)':<55} {'Match?'}")
    for name, out in rows:
        rounded = [round(float(v), 4) for v in out.flatten()]
        ok = np.allclose(out, expected)
        if name == "TFLite" and not ok:
            any_bug = True
        print(f"  {name:<14} {str(rounded):<55} {'OK' if ok else 'WRONG'}")
    print()

sys.exit(0 if any_bug else 1)
```

optput

  expected : [0.0699, 0.1049, 0.1399, 0.1748, 0.2098, 0.2447, 0.2797, 0.3147, 0.3496, 0.3846, 0.4196, 0.4545]
  Keras eager    matches expected                                              OK
  XLA            matches expected                                              OK
  PyTorch        matches expected                                              OK
  ONNX Runtime   matches expected                                              OK
  TFLite         [0.3714, 0.5571, 0.7428, 0.4767, 0.5721, 0.6674, 0.5111, ...] WRONG (each innermost-3 vec normalized independently)

  expected : [0.2722, 0.4082, 0.5443, 0.6804]
  Keras eager / XLA / PyTorch / ONNX Runtime: matches expected                 OK
  TFLite         [1.0, 1.0, 1.0, 1.0]                                          WRONG (each scalar / itself = 1.0)
BUG REPRODUCED on both cases — confirms data-independence of the axis-semantics mismatch

**Comment by aravindmurali22** [comment]
Hi @beanduan22 ,
Thank you for taking the time to report this and for your patience.
TFLite has [transitioned](https://developers.googleblog.com/en/tensorflow-lite-is-now-litert/) to LiteRT, Google's primary runtime for on-device AI. All active development, feature updates, and performance enhancements have shifted  to LiteRT. We encourage you to try running your model with LiteRT. If you still encounter this issue, please open a new issue with the updated details in the [LiteRT Repository](https://github.com/google-ai-edge/LiteRT).
For more information and step-by-step guidance on updating your project, please refer to the [LiteRT Migration Guide](https://developers.google.com/edge/litert/migration).
Thank you again!
